sympy/core/mul.py

```diff
@@ -1255,19 +1255,20 @@ def _eval_is_integer(self):
             return False
 
         n, d = fraction(self)
-        if not is_rational:
-            _self = n/d
-            if _self != self:
-                return _self.is_integer
         if is_rational:
-            if d is S.One:
+            if d == S.One:
```

Comment:
`d` can never be 1.0 so if it equals 1 it is S.One

Reply:
Ohhh okey! gotcha